onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -341,8 +341,9 @@
       // currently the only level 3 optimizer is the NhwcTransformer which is fully supported at runtime
       if (!saving) {
 #ifndef DISABLE_CONTRIB_OPS
+        const InlinedHashSet<std::string_view> cpu_ep = {onnxruntime::kCpuExecutionProvider};
         auto cpu_allocator = cpu_execution_provider.GetAllocator(0, OrtMemTypeDefault);
-        transformers.emplace_back(std::make_unique<NhwcTransformer>(std::move(cpu_allocator)));
+        transformers.emplace_back(std::make_unique<NhwcTransformer>(cpu_ep, std::move(cpu_allocator)));
 #else
```

Comment:
This is a good example of why it's better for the EP to do its own optimizations. The core ORT code should not need to be aware of any non-ORT EP requirements as that doesn't scale.

Reply:
Level 2 is EP dependent optimization. Level 3 is for CPU EP ARCH dependent optimization. That's why EP is hardcoded in the transformer.

I just traced the code history. NhwcTransformer also had CPU EP check previously, but was removed in this PR:
[https://github.com/microsoft/onnxruntime/pull/9284/files#diff-930bf0c155479c9a2e17b4996b0ae133e7db12e3161b6ac868e6e9756d76771bL486-L488](https://github.com/microsoft/onnxruntime/pull/9284/files#diff-930bf0c155479c9a2e17b4996b0ae133e7db12e3161b6ac868e6e9756d76771bL486-L488)